src/PlaylistSave.cxx

```diff
@@ -83,17 +85,44 @@ playlist_print_uri(BufferedOutputStream &os, const char *uri)
 	}
 }
 
+PlaylistSaveMode
+PlaylistSaveMode::parseArg(const char* arg) {
+	PlaylistSaveMode mode;
+	if (arg == nullptr) {
+		mode.create = true;
+	} else {
+		if (StringIsEqual(arg, "create"))
+			mode.create = true;
+		else if (StringIsEqual(arg, "append"))
+			mode.append = true;
+		else if (StringIsEqual(arg, "replace"))
+			mode.replace = true;
+	}
+
+	if (!mode.is_valid())
+		throw MakeArgError(mode.argValidationError, arg == nullptr ? "" : arg);
```

Comment:
This whole protocol-specific method living in a non-protocol-specific library is kind of a layering violating - I suggest we leave it that way for now, but please use `std::invalid_argument`; if caught by a command handler, it will be translated to ACK_ERROR_ARG, so that's fine:
https://github.com/MusicPlayerDaemon/MPD/blob/512cd7b0dea47883e54245122171a0178678e178/src/command/CommandError.cxx#L99-L100